### PR TITLE
Reduce Helix keyboard build variation

### PR DESCRIPTION
**Pull request by MakotoKurauchi**
## Description

There are many build variation for the current Helix keyboard. It was a little too much, so I chose patterns that users often use and reduced the unnecessary ones.

**This change does not change the build results.**

<details>
<summary><strong>All compile option patterns (click)</strong></summary>

* build helix/pico (HelixPico) with helix current codes
  ```
  $ make helix/pico:KEY_MAP
  $ make helix/pico/back:KEY_MAP
  $ make helix/pico/under:KEY_MAP
  ```

* build helix/rev2 (Helix or Helix beta) with helix current codes
  Allways OLED enable.
  ```
  $ make helix:KEY_MAP
  $ make helix/rev2/back:KEY_MAP
  $ make helix/rev2/under:KEY_MAP
  ```

* build helix/pico (HelixPico) with split_common codes
  RGBlight as backlight enable.
  ```
  $ make helix/pico/split_common:KEY_MAP
  ```
  
* build helix/rev2 (Helix) with split_common codes
  OLED and RGBlight as backlight enable.
  ```
  $ make helix/rev2/split_common:KEY_MAP
  ```
</details>
<details>
<summary><strong>Comparison of old and new (click)</strong></summary>

| build name in the previous version | build name in the new version |
| ---------------------------- | ------------------------|
| helix/rev1                   | helix/rev1              |
| helix/pico                   | helix/pico              |
| helix/pico/back              | helix/pico/back         |
| helix/pico/under             | helix/pico/under        |
| helix/pico/sc                | --                      |
| helix/pico/sc/back           | helix/pico/split_common |
| helix/pico/sc/under          | --                      |
| helix/rev2 (=helix)          | --                      |
| helix/rev2/back              | --                      |
| helix/rev2/back/oled         | --                      |
| helix/rev2/oled              | helix/rev2　(=helix)    |
| helix/rev2/oled/back         | helix/rev2/back         |
| helix/rev2/oled/under        | helix/rev2/under        |
| helix/rev2/sc                | --                      |
| helix/rev2/sc/back           | --                      |
| helix/rev2/sc/oled           | --                      |
| helix/rev2/sc/oledback       | helix/rev2/split_common |
| helix/rev2/sc/oledunder      | --                      |
| helix/rev2/sc/under          | --                      |
| helix/rev2/under             | --                      |
| helix/rev2/under/oled        | --                      |
</details>

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/helix/rev2/keymaps/xulkal/rules.mk [hunk 1]
```diff
@@ -11,3 +11,6 @@ OPT_DEFS += -DOLED_FONT_H=\"common/glcdfont.c\"
 OPT_DEFS += -DOLED_90ROTATION
 
 SPLIT_KEYBOARD = yes
+
+# Explicitly disable helix standard post-processing
+KEYBOARD_LOCAL_FEATURES_MK :=
```
Comment (mtei) [inline keyboards/helix/rev2/keymaps/xulkal/rules.mk:16]: @XScorpion2, Your confirmation is needed.

Comment (XScorpion2) [reply keyboards/helix/rev2/keymaps/xulkal/rules.mk:16]: approved

Comment (mtei) [reply keyboards/helix/rev2/keymaps/xulkal/rules.mk:16]: Thank you !

**Comment by mtei** [comment]
It's only recently that build variations have increased. The main purpose of this PR is to bring it back because it was increased too much.
(I'll explain more about the history later if require.)

Does this require the breaking change process? I don't think so.

| before #6952                 | 881f27b461 #6952 Oct 17 2019 | 2d14d12c74 #7718 Mar 10 2020 | this PR |
| ---------------------------- | ---------------------------- | ---------------------------- | ------------------------|
| helix/rev1                   | helix/rev1                   | helix/rev1                   | helix/rev1              |
| helix/pico                   | helix/pico                   | helix/pico                   | helix/pico              |
|                              | helix/pico/back              | helix/pico/back              | helix/pico/back         |
|                              | helix/pico/under             | helix/pico/under             | helix/pico/under        |
|                              |                              | helix/pico/sc                | --                      |
|                              |                              | helix/pico/sc/back           | helix/pico/split_common |
|                              |                              | helix/pico/sc/under          | --                      |
| helix/rev2 (=helix)          | helix/rev2 (=helix)          | helix/rev2 (=helix)          | --                      |
|                              | helix/rev2/back              | helix/rev2/back              | --                      |
|                              | helix/rev2/back/oled         | helix/rev2/back/oled         | --                      |
|                              | helix/rev2/oled              | helix/rev2/oled              | helix/rev2　(=helix)    |
|                              | helix/rev2/oled/back         | helix/rev2/oled/back         | helix/rev2/back         |
|                              | helix/rev2/oled/under        | helix/rev2/oled/under        | helix/rev2/under        |
|                              |                              | helix/rev2/sc                | --                      |
|                              |                              | helix/rev2/sc/back           | --                      |
|                              |                              | helix/rev2/sc/oled           | --                      |
|                              |                              | helix/rev2/sc/oledback       | helix/rev2/split_common |
|                              |                              | helix/rev2/sc/oledunder      | --                      |
|                              |                              | helix/rev2/sc/under          | --                      |
|                              | helix/rev2/under             | helix/rev2/under             | --                      |
|                              | helix/rev2/under/oled        | helix/rev2/under/oled        | --                      |

**Comment by zvecr** [comment]
The what and why of this PR is not where the issue lies, that part is fairly obvious from the diffs.

The issue is that if a end user was previously building with `helix/pico/sc`, there is no indication that they now have to build with `helix/pico/split_common` instead.

**Comment by mtei** [comment]
Are you saying that the renaming of `rev2/sc` to `rev2/split_common` is a bad idea? Is it better to stop renaming `rev2/sc`? If so, I will stop the name change.

Or does it mean it's not good that the target that existed is no longer there? (For example, `rev2/oled/back` has been changed to `rev2/back`.)

**Comment by mtei** [comment]
Don't overlook that there are only three weeks between the merging of the last version (#7718) and the opening of this PR. This PR is effectively a bug fix for the previous version.

The `readme.md` explains that `helix/rev2/sc` is an `(experimental use of split_common)`. Users would normally avoid using this target.

**Comment by noroadsleft** [comment]
Checking if I understand correctly what this PR does if merged:

- `helix/rev1` = unchanged
- `helix/pico/sc`, `helix/pico/sc/back` and `helix/pico/sc/under` are now together under `helix/pico/sc`, and the user changes the `make` command based on what options they want (LED backlight or LED underglow).
  - The non-split_common `helix/pico` variants are unchanged
- `helix/rev2` - OLED enabled becomes default for both split_common and non-split_common variants. Further build targets configure whether LED backlight or LED underglow is enabled, similarly to `helix/pico`

Do I have this right?

**Comment by mtei** [comment]
> * `helix/rev1` = unchanged

Yes.

> * `helix/pico/sc`, `helix/pico/sc/back` and `helix/pico/sc/under` are now together under `helix/pico/sc`, and the user changes the `make` command based on what options they want (LED backlight or LED underglow).

The split_common `helix/pico/sc` variants are experimental targets that were added at the time of #7718. Since it's experimental, MakotoKurauchi and I have yet to recommend it to the average user. That's why we reduced the variations.

We reduced the variation of split_common `helix/rev2/sc` for the same reason.

>   * The non-split_common `helix/pico` variants are unchanged

Yes.

> * `helix/rev2` - OLED enabled becomes default for both split_common and non-split_common variants. Further build targets configure whether LED backlight or LED underglow is enabled, similarly to `helix/pico`

Yes. MakotoKurauchi's shop sells the Helix kit with OLED included as standard. That's why we've made OLEDs the default.

**Comment by noroadsleft** [comment]
Thanks, @mtei. I've spent quite some time (many months) mostly working on Configurator, so I don't remember my way around makefiles anymore. :/

**Comment by Erovia** [comment]
@MakotoKurauchi, @mtei what do you think about retargeting this PR to the `develop` branch and getting it merged there first?

**Comment by mtei** [comment]
I don't think it makes any difference whether this PR is merged into master or into develop.

Because, as I already mentioned above, I don't think any of our users are using the target to be erased by this PR.

Therefore, the answer is weak approval.

**Pull request merged**